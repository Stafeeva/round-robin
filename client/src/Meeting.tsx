import React, { FC, useEffect } from "react";
import { io } from "socket.io-client";
import { Avatar, Button, Flex, Skeleton } from "antd";
import {
  UserOutlined,
  ClockCircleFilled,
  AudioFilled,
} from "@ant-design/icons";

import { Attendee, Action, Note } from "../../server/server";
import "./App.css";
import { AddAction } from "./App";
import NoteItem from "./components/Note";
import AddNote from "./components/AddNote";
import ActionItem from "./components/ActionItem";

const socket = io();

const Meeting = ({ attendeeName }: { attendeeName: string }) => {
  const [meeting, setMeeting] = React.useState<any>({});

  const [notes, setNotes] = React.useState<Note[]>([]);

  const [actions, setActions] = React.useState<any[]>([]);

  const fetchMeeting = async () => {
    const res = await fetch("/api/meeting");
    const data = await res.json();
    setMeeting(data);
  };

  const fetchNotes = async () => {
    const res = await fetch("/api/note");
    const data = await res.json();
    setNotes(data);
  };

  const fetchActions = async () => {
    const res = await fetch("/api/action");
    const data = await res.json();
    setActions(data);
  };

  useEffect(() => {
    // fetchMeeting();
    fetchNotes();
    // fetchActions();

    // TODO: type for data
    // socket.on("meetingUpdated", (data: any) => {
    //   setMeeting(data);
    // });

    socket.on("noteAdded", (data: Note[]) => {
      setNotes(data);
    });

    // socket.on("actionAdded", (data: Action[]) => {
    //   setActions(data);
    // });
  }, []);

  const handleStartMeeting = async () => {
    await fetch("/api/meeting:start", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
    });
  };

  const handleNext = async () => {
    await fetch("/api/meeting:next", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
    });
  };

  const handleResetMeeting = async () => {
    await fetch("/api/meeting:reset", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
    });
  };

  return (
    <Flex
      vertical
      style={{ width: "100%", height: "100vh" }}
      align="center"
      justify="flex-start"
    >
      <Flex
        vertical
        style={{ width: "800px", padding: "48px" }}
        align="center"
        gap="48px"
      >
        {/* <h2>Meeting: {meeting?.meetingState}</h2>
          < */}
        <h2>Apollo team standup</h2>
        {/* {meeting?.meetingState === "InProgress" && (
            <div className="speakers">
              <p>
                <strong>Current Speaker: </strong>{" "}
                {meeting?.currentSpeaker?.name}
              </p>
              {meeting?.speakerQueue?.length > 0 && (
                <p>
                  <strong>Speaker queue</strong>:{" "}
                  {meeting?.speakerQueue?.map(
                    (attendee: Attendee, i: number) => (
                      <span className="speaker" key={i}>
                        {attendee.name}
                      </span>
                    )
                  )}
                </p>
              )}
            </div>
          )} */}

        <div className="meeting-actions">
          {meeting?.meetingState === "NotStarted" ? (
            <Button onClick={handleStartMeeting}>Start</Button>
          ) : (
            <Button onClick={handleResetMeeting}>Reset</Button>
          )}
          {meeting?.meetingState === "InProgress" && (
            <Button onClick={handleNext}>
              {meeting?.speakerQueue?.length === 0 ? "Finish" : "Next"}
            </Button>
          )}
        </div>

        <div style={{ width: "100%" }}>
          {meeting.meetingState === "InProgress" ? (
            <Flex justify="space-between">
              <div>
                <ClockCircleFilled style={{ fontSize: "48px" }} />
                <span style={{ fontSize: "48px", marginLeft: "24px" }}>
                  01:18
                </span>
              </div>
              <div>
                <AudioFilled style={{ fontSize: "48px" }} />
                <span style={{ fontSize: "48px", marginLeft: "24px" }}>
                  Dylan
                </span>
              </div>
            </Flex>
          ) : null}
        </div>

        <div className="attendees">
          <h2>Attendees</h2>
          <Flex gap="24px" style={{ marginTop: "24px" }}>
            {meeting?.attendees?.map((attendee: Attendee, i: number) => (
              <div
                key={i}
                className={attendeeName === attendee.name ? "active" : ""}
              >
                <Avatar size={64} icon={<UserOutlined />} />
                <div>{attendee.name}</div>
              </div>
            ))}
          </Flex>
        </div>
      </Flex>
      <Flex style={{ width: "800px" }} gap="48px">
        <div className="notes">
          <h2>Notes</h2>
          <AddNote author={attendeeName} />
          <ul>
            {notes?.map((note: Note, i: number) => (
              <NoteItem note={note} key={`note-${i}`} />
            ))}
          </ul>
        </div>

        <div className="actions">
          <h2>Actions</h2>
          <AddAction attendees={meeting?.attendees || []} />
          {/* {actions?.map((action, i: number) => ( */}
          <ActionItem
            actionItem={{
              asignee: "Dylan",
              text: "This is an action item",
              completed: false,
            }}
          />
          {/* ))} */}
        </div>
      </Flex>
    </Flex>
  );
};

export default Meeting;
