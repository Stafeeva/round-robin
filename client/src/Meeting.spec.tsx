import React from "react";
import { render } from "@testing-library/react";

import Meeting from "./Meeting";

const meeting = () => render(<Meeting attendeeName="Dylan" />);

test("Meeting component renders", () => {
  const { container } = meeting();
  expect(container).toMatchSnapshot();
});

// test("User can start a meeting by clicking Start button", () => {
//   const { getByText } = meeting();
//   const startButton = getByText("Start");
//   expect(startButton).toBeInTheDocument();
// });
