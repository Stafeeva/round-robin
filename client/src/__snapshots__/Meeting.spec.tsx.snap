// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meeting component renders 1`] = `
<div>
  <div
    class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-flex-start ant-flex-vertical"
    style="width: 100%; height: 100vh;"
  >
    <div
      class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-vertical"
      style="width: 800px; padding: 48px; gap: 48px;"
    >
      <h2>
        Apollo team standup
      </h2>
      <div
        class="meeting-actions"
      >
        <button
          class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default"
          type="button"
        >
          <span>
            Reset
          </span>
        </button>
      </div>
      <div
        style="width: 100%;"
      />
      <div
        class="attendees"
      >
        <h2>
          Attendees
        </h2>
        <div
          class="ant-flex css-dev-only-do-not-override-1uq9j6g"
          style="margin-top: 24px; gap: 24px;"
        />
      </div>
    </div>
    <div
      class="ant-flex css-dev-only-do-not-override-1uq9j6g"
      style="width: 800px; gap: 48px;"
    >
      <div
        class="notes"
      >
        <h2>
          Notes
        </h2>
        <div
          class="ant-skeleton ant-skeleton-with-avatar css-dev-only-do-not-override-1uq9j6g"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
              <li />
              <li />
            </ul>
          </div>
        </div>
        <div
          class="ant-skeleton ant-skeleton-with-avatar css-dev-only-do-not-override-1uq9j6g"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
            </ul>
          </div>
        </div>
        <div
          class="ant-skeleton ant-skeleton-with-avatar css-dev-only-do-not-override-1uq9j6g"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
              <li />
            </ul>
          </div>
        </div>
      </div>
      <div
        class="actions"
      >
        <h2>
          Actions
        </h2>
        <div
          class="ant-skeleton ant-skeleton-with-avatar css-dev-only-do-not-override-1uq9j6g"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
            </ul>
          </div>
        </div>
        <div
          class="ant-skeleton ant-skeleton-with-avatar css-dev-only-do-not-override-1uq9j6g"
        >
          <div
            class="ant-skeleton-header"
          >
            <span
              class="ant-skeleton-avatar ant-skeleton-avatar-lg ant-skeleton-avatar-circle"
            />
          </div>
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 50%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
              <li />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
